bot/code_review_bot/__init__.py

```diff
@@ -127,6 +127,19 @@ def build_extra_identifiers(self):
         """
         return {}
 
+    @property
+    def allow_before_and_after(self):
+        """
+        Allow the possibility for an issue to avoid being published based on before/after.
+        This allow publishing issues based on other criteria, like in_patch.
+        """
+        if taskcluster.secrets.get(
```

Comment:
I think this condition should rather be implemented in `code_review_bot.revisions.Revision.before_after_feature`, so it also disable the feature in `code_review_bot.workflow`. Otherwise some issues will not be published but still analyzed with the feature and so fail comparing with previous issues.

Reply:
After looking closely at the publication part, this should work as expected until those settings changes between the base revision and the analyzed revision (issues reported as new, or closed with before/after because not published in the backend).
The workflow is:
Issue detected -> marked as new if not published before -> not published -> not mentioned on Phabricator
Ideally it would be:
Issue detected -> marked as new if not published before -> published -> not mentioned on Phabricator

Otherwise I suggest to simply rename `<ANALYZER>_DISABLE_BEFORE_AFTER` to `<ANALYZER>_DISABLE_PUBLICATION` to make it clear. The check can be done in the `is_publishable` method directly also.